pennylane/transforms/decompose.py

```diff
@@ -0,0 +1,210 @@
+    """
+    return results[0]
+
+
+@transform
+def decompose(tape, gate_set=None, max_expansion=None):
+    """Decomposes a quantum circuit into a user-specified gate set.
+
+    Args:
+        tape (QuantumScript or QNode or Callable): a quantum circuit.
+        gate_set (Iterable[Union[str, type]] or Callable[Operator, bool], optional): Decomposition gates defined by either (1) a gate set of operators or (2) a rule that they must follow.
+        Defaults to None. If ``None``, gate set defaults to all available operators given by ``~.pennylane.ops.__all__``.
+        max_expansion (int, optional): The maximum depth of the decomposition. Defaults to None. If ``None``, circuit will be decomposed until the target gate set is reached.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumScript], function]:
+
+        The decomposed circuit. The output type is explained in :func:`qml.transform <pennylane.transform>`.
+
+    .. seealso:: :func:`~.pennylane.devices.preprocess.decompose` for a transform that is intended for device developers. This function
+    decomposes a quantum circuit into a set of basis gates available on a given device architecture.
+
+    **Example**
+
+    Consider the following tape:
+
+    >>> tape = qml.tape.QuantumScript([qml.IsingXX(1.2, wires=(0,1))], [qml.expval(qml.Z(0))])
+
+    You can then decompose the circuit into a set of gates:
+
+    >>> batch, fn = decompose(tape, gate_set={qml.CNOT, qml.RX})
+    >>> batch[0].circuit
+    [CNOT(wires=[0, 1]), RX(1.2, wires=[0]), CNOT(wires=[0, 1]), expval(Z(0))]
+
+    You can also apply the transform directly on a :class:`QNode`:
+
+    .. code-block:: python
+
+        @partial(decompose, gate_set={qml.Toffoli, "RX", "RZ"})
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit():
+            qml.Hadamard(wires=[0])
+            qml.Toffoli(wires=[0,1,2])
+            return qml.expval(qml.Z(0))
+
+    Since the Hadamard gate is not defined in our gate set, it will be decomposed into rotations:
+
+    >>> print(qml.draw(circuit)())
+    0: ──RZ(1.57)──RX(1.57)──RZ(1.57)─╭●─┤  <Z>
+    1: ───────────────────────────────├●─┤
+    2: ───────────────────────────────╰X─┤
+
+    You can also use callable functions to build a decomposition gate set:
+
+    .. code-block:: python
+
+        @partial(decompose, gate_set=lambda op: len(op.wires) <= 2)
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit():
+            qml.Hadamard(wires=[0])
+            qml.Toffoli(wires=[0,1,2])
+            return qml.expval(qml.Z(0))
+
+    In this case, the circuit will be decomposed into operators acting on no more than two wires:
+
+    >>> print(qml.draw(circuit)())
+    0: ──H────────╭●───────────╭●────╭●──T──╭●─┤  <Z>
+    1: ────╭●─────│─────╭●─────│───T─╰X──T†─╰X─┤
+    2: ──H─╰X──T†─╰X──T─╰X──T†─╰X──T──H────────┤
+
+    You can use the 'max_expansion' kwarg to have control over the number
```

Comment:
```suggestion
    You can use the ``max_expansion`` kwarg to have control over the number
```
Same as above. this should probably have the code formatting back ticks:

<img width="351" alt="Screenshot 2024-10-08 at 4 54 34 PM" src="https://github.com/user-attachments/assets/7cfe6291-b64c-4d69-bc4d-da583a588b92">